### ISSUE TITLE
ExtAuth failOnRedirect doesn't work

### ISSUE THREAD

**Issue by bcollard** [solo-io/gloo#7167]
### Gloo Edge Version

1.11.x

### Kubernetes Version

1.23.x

### Describe the bug

Given an AuthConfig with two steps, as in:

```yaml
apiVersion: enterprise.gloo.solo.io/v1
kind: AuthConfig
metadata:
  name: keycloak-oauth
  namespace: gloo-system
spec:
  booleanExpr: "(oidc || passthrough)"
  configs:
  - name: oidc
    oauth2:
      oidcAuthorizationCode:
        appUrl: https://mydomain.com
        callbackPath: /callback
        clientId: dd99ff5a-5e5f-41ac-b080-7136e43f8e8c
        clientSecretRef:
          name: keycloak-oauth
          namespace: gloo-system
        issuerUrl: http://172.18.1.2:8080/auth/realms/master/
        scopes:
        - email
  - name: passthrough
    passThroughAuth:
      http:
        url: http://http-echo.default.svc:8080
  failOnRedirect: true
```

If the end-user is not logged in, then Keycloack will return a 302 to the login page.
We expect the `failOnRedirect: true` option to prevent this behavior and return a 401 instead.
This is not happening.

### Steps to reproduce the bug

see above

### Expected Behavior

401 HTTP status

**Comment by sirrapa** [comment]
Hi,
Is there any progress on this issue?

**Comment by sirrapa** [comment]
Hi,

Is there any update on this issue

**Comment by sam-heilbron** [comment]
PRs that initially introduced this feature:
- https://github.com/solo-io/gloo/pull/6844
- https://github.com/solo-io/ext-auth-service/pull/375
- https://github.com/solo-io/solo-projects/pull/3936

**Comment by sirrapa** [comment]
Hi @sam-heilbron,

I've updated to 1.15.5 but it still is not returning a 401.
The way I test it is by doing a curl to a endpoint which is configured with an AuthConfig.
```
spec:
  booleanExpr: (oauth2 || passThroughAuth || opaAuth)
  configs:
  - oauth2:
      oidcAuthorizationCode:
        appUrl: https://***app.nl
        callbackPath: /2fa/callback
        clientId: *******-be59-4ff1-813c-*********
        clientSecretRef:
          name: **secret**
          namespace: gloo-custom
        headers:
          accessTokenHeader: X-Jwt
          idTokenHeader: jwt
        issuerUrl: https://**********/adfs
        logoutPath: /logout
        scopes:
        - email
        - groups
        - openid
        - profile
        session:
          cookieOptions:
            maxAge: 3600
          redis:
            allowRefreshing: true
            cookieName: cmMonster
            options:
              host: redis.gloo-system.svc.cluster.local:6379
  - passThroughAuth:
      grpc:
        address: extauth-server.svc.cluster.local:9001
        connectionTimeout: 3s
  - opaAuth:
      modules:
      - name: glooe-rbac-jwt
        namespace: gloo-system
      query: data.app.rbac.allow == true
  failOnRedirect: true

**Comment by SantoDE** [comment]
@sam-heilbron just had a conversation with @sirrapa around that. It happens even if its just one Auth in the config. Let's sync on that Monday.

**Comment by sam-heilbron** [comment]
This work will be available in the following Enterprise releases:
- 1.15.9
- 1.16.0-beta3

**Comment by sam-heilbron** [comment]
The work to introduce this fix into the 1.16 branch can be found here: https://github.com/solo-io/solo-projects/pull/5542#issuecomment-1851980288
